### PR TITLE
PEP257 Clean-up of docstrings.

### PR DESCRIPTION
**Pull request by dhermes**
Fixes all Pylint failures of 
- `missing-returns-doc`
- `redundant-returns-doc`
- `ungrouped-imports`

Also makes a little headway on `missing-raises-doc` though it's unclear if this error should be fixed. Also https://github.com/PyCQA/pylint/commit/8d355577a0bc28c8ca6d94ec4723b7e9b818c047 makes the bad `catching-non-exception` go away.

Towards #1968.

### CODE REVIEW

File: gcloud/_helpers.py [hunk 1]
```diff
@@ -439,8 +441,7 @@ def _to_bytes(value, encoding='ascii'):
     :rtype: str / bytes
     :returns: The original value converted to bytes (if unicode) or as passed
               in if it started out as bytes.
-    :raises: :class:`TypeError <exceptions.TypeError>` if the value
-             could not be converted to bytes.
+    :raises TypeError: if the value could not be converted to bytes.
```
Comment (dhermes) [inline gcloud/_helpers.py]: @tseaver This is what `pylint` requires to "believe" that `TypeError` is mentioned in the `Raises` section. Unfortunately, the Sphinx generated docs never turn this into a link (not a big deal for stdlib exceptions, but it's nice to have links for custom exceptions)

Comment (tseaver) [inline gcloud/_helpers.py]: Does it allow for the `exc` role in the text of `:raises:`?  e.g.

```
`:raises: :exc:`TypeError`
```

I find that syntax preferable to `:class:` for exceptions.

Comment (dhermes) [inline gcloud/_helpers.py]: I'm not sure. I played around a bit and didn't find many forms valid to Sphinx that `pylint` was happy with.

**Comment by dhermes** [comment]
@tseaver FYI I added another commit since `pylint` 1.6.4 fixed a false positive.

File: gcloud/_helpers.py (hunk 1 above)
Comment (tseaver) [inline gcloud/_helpers.py]: I'd prefer something semantically richer, and that rendered well in Sphinx, over something that appeases pylint.  What if we ask the pylint team to add support for

```
:raises: :exc:`ValueError`
```

Comment (dhermes) [inline gcloud/_helpers.py]: Yeah totally. For now I'm not touching `missing-raises-doc` in this PR other than the 3 or 4 places I modified it to understand what was triggering the error / how to fix it.

**Comment by dhermes** [comment]
@tseaver Issues remaining?

**Comment by tseaver** [comment]
LGTM

**Pull request merged**